pkg/frontend/mysql_cmd_executor.go

```diff
@@ -2158,6 +2179,94 @@ func (cwft *TxnComputationWrapper) Compile(requestCtx context.Context, u interfa
 	if err != nil {
 		return nil, err
 	}
+	// check if it is necessary to initialize the temporary engine
+	if cwft.compile.NeedInitTempEngine(cwft.ses.IfNeedInitTempEngine()) {
+		// 0. init memory-non-dist engine
+		ck := clock.DefaultClock()
+		if ck == nil {
+			ck = clock.NewHLCClock(func() int64 {
+				return time.Now().Unix()
+			}, math.MaxInt)
+		}
+		shard := logservicepb.DNShardInfo{
+			ShardID:   2,
+			ReplicaID: 2,
+		}
+		shards := []logservicepb.DNShardInfo{
+			shard,
+		}
+		dnAddr := "1"
+		dnStore := logservicepb.DNStore{
+			UUID:           uuid.NewString(),
+			ServiceAddress: dnAddr,
```

Comment:
and "1"?   what is it?

Reply:
same as "2"

Reply:
Any value is OK here. I suggest "DN-address".